polaris/competition/_base.py

```diff
@@ -0,0 +1,229 @@
+    that in combination with the [`DatasetV2`][polaris.experimental._dataset_v2.DatasetV2] class, allow
+    users to participate in competitions hosted on Polaris Hub.
+
+    Examples:
+        Basic API usage:
+        ```python
+        import polaris as po
+
+        # Load the benchmark from the Hub
+        competition = po.load_competition("polaris/hello-world-competition")
+
+        # Get the train and test data-loaders
+        train, test = competition.get_train_test_split()
+
+        # Use the training data to train your model
+        # Get the input as an array with 'train.inputs' and 'train.targets'
+        # Or simply iterate over the train object.
+        for x, y in train:
+            ...
+
+        # Work your magic to accurately predict the test set
+        prediction_values = np.array([0.0 for x in test])
+        predictions = CompetitionPredictions(
+            name="first-prediction",
+            owner=HubOwner(slug="dummy-user"),
+            report_url="REPORT_URL",
+            target_labels=competition.target_cols,
+            test_set_labels=competition.test_set_labels,
+            test_set_sizes=competition.test_set_sizes,
+            predictions=prediction_values,
+        )
+
+        # Submit your predictions
+        competition.submit_predictions(predictions)
+        ```
+
+    Attributes:
+        target_cols: The column(s) of the original dataset that should be used as target.
+        input_cols: The column(s) of the original dataset that should be used as input.
+        split: The predefined train-test split to use for evaluation.
+        metrics: The metrics to use for evaluating performance
+        main_metric: The main metric used to rank methods.
+        readme: Markdown text that contains a formatted description of the competition.
+        target_types: A dictionary that maps target columns to their type.
+        start_time: The time at which the competition starts accepting prediction submissions.
+        end_time: The time at which the competition stops accepting prediction submissions.
+        n_test_sets: The number of test sets defined in the test split.
+        n_test_datapoints: The number of datapoints in each test set.
+        n_classes: The number of classes within target columns that define a classification task.
+    For additional meta-data attributes, see the [`DatasetV2`][polaris.experimental._dataset_v2.DatasetV2] class.
+    """
+
+    _artifact_type = "competition"
+
+    target_cols: ColumnsType
+    input_cols: ColumnsType
+    split: SplitType
+    metrics: Union[str, Metric, list[str | Metric]]
+    main_metric: str | Metric
+    readme: str
+    target_types: dict[str, Union[TargetType, str]] = Field(default_factory=dict, validate_default=True)
+    start_time: datetime
+    end_time: datetime
+    n_test_sets: int
+    n_test_datapoints: dict[str, int]
+    n_classes: dict[str, int]
```

Comment:
**Comment**: Let's not do this now, but I do think we'll want to refactor this to inherit from some sort of `BaseBenchmark` or `BenchmarkMetadata` class. There's a lot of code duplication between this class and the benchmark class.

Let's revisit that once the `BenchmarkV2` is ready!

Reply:
Agreed 👍